enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,910 @@
+
+It is also feasible that a customer may want to remove the Machine API resources after they have migrated to Cluster API and no longer require the Machine API synchronisation.
+To allow for this, if a non-authoritative Machine API parent[^8] resource is deleted, the deletion event will not be synchronised to the Cluster API equivalent and the synchronisation controller will ensure finalizers are removed as appropriate.
+In this scenario, the Cluster API equivalent will continue to operate on its own.
+To ensure users are aware of this situation, webhooks will be introduced to warn users that the deletion will not be synchronised.
+
+[^8]: To allow continued operation of MachineSets, ControlPlaneMachineSets and MachineHealthChecks, if a Machine API Machine that is owned is deleted,
+then the deletion event is synchronised to Cluster API.
+This will allow for a MachineSet/ControlPlaneMachineSet to own mixed instances (i.e. those managed by both Machine API and Cluster API), and continue to support higher level remediation automation.
+To remove the Machine API resources, the user should delete the parent, i.e. the MachineSet or ControlPlaneMachineSet on the Machine API side.
+
+#### Scenarios with mixed Machines
+
+The following scenarios explain the logical flow of what must happen if any of the controllers are monitoring a selection of Machines where the authoritative API is mixed between Cluster API and Machine API.
+
+When the selection of Machines all match the same API, the normal flows will be observed.
+
+##### MachineSet scale up
+
+If a MachineSet scales up, it will create a new Machine as it does today.
+Because the managed resource in this scenario is the MachineSet the sync controller will ensure that the Machine resources are reflected in both Machine API and Cluster API.
+
+If the MachineSet is a Cluster API MachineSet and there is no Machine API equivalent, this synchronisation will not happen and the machines will be created as Cluster API only.
+
+New Machines created in the scale up will default to the same authoritative API as the MachineSet that creates them.
+
+##### MachineSet scale down
+
+If a MachineSet scales down, this will trigger the deletion of a Machine resource
+To ensure consistency between the two MachineSets, the sync controller will synchronise the delete event between the Cluster API and Machine API Machines.
+
+##### MachineHealthCheck remediates Machine
+
+As MachineHealthChecks only remediate Machines that have owners, when the deletion event occurs, the deletion event will be synchronised across between the two APIs.
+
+##### Resource created as Cluster API Resource, wish to migrate to Machine API Resource
+
+To enable a reverse migration, should the customer wish, we will need to enable the user to create a Machine API equivalent of a Cluster API resource if it does not exist.
```

Comment:
Within the `openshift-cluster-api` namespace, we expect the resources here to only be for the cluster itself and not for any guest clusters.

Since we don't today support multiple platforms for a cluster, there should be no reason to have a non-matching Infrastructure template.

As for reasons for this, we expect customers are not going to want to have to deal with two APIs at once, for an extended period. If they test the infrastructure migration and decide they've hit a roadblock, we want to allow them to go backwards. Given other parts of this enhancement that allow removing MAPI equivalent objects for existing CAPI resources, this is the counter to that and the get out of jail free card